go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -209,6 +210,15 @@ func (a *PodAllocator) releasePodOnNAD(pod *corev1.Pod, nad string, podDeleted,
 
 	hasIPAM := util.DoesNetworkRequireIPAM(a.netInfo)
 	hasIDAllocation := util.DoesNetworkRequireTunnelIDs(a.netInfo)
+	hasPersistentIPs := network.IPAMClaimReference != ""
+	if hasPersistentIPs {
+		ipamClaim, err := a.findIPAMClaim(network)
+		if err != nil {
+			hasPersistentIPs = false
```

Comment:
Don't we have to filter NotFound ? we should fail on other errors or at least log them.

Reply:
if we're being requested to use persistent IPs the client must have created these before. 

Not finding the IPAMClaim is an error leading to a retry with exponential backoff. Any argument for a special behavior here ? I don't see reasons for it.

Reply:
we may have other kind of errors when accessing api server

Reply:
Nah all good